src/cast_core/src/CollectionScanner.cpp

```diff
@@ -161,29 +234,47 @@ void CollectionScanner::run() {
                 continue;
             }
             _runningActorCounter++;
-            BOOST_LOG_TRIVIAL(info) << "Starting collection scanner id: " << this->_index;
-
-            // Populate collections if need be.
-            std::vector<mongocxx::collection>& collections = config->collections;
-            if (config->queryCollectionList) {
-                std::vector<mongocxx::collection> tempCollections{};
-                for (const auto& collection : config->database.list_collection_names({})) {
-                    tempCollections.push_back(config->database[collection]);
+            BOOST_LOG_TRIVIAL(info) << "Starting collection scanner databases: \"" << _databaseCommaList << "\", id: " << this->_index;
+
+            // Populate collection names if need be.
+            std::vector<mongocxx::collection> collections;
+            for (auto database : config->databases) {
+                if (config->queryCollectionList) {
+                    config->collectionsFromNameList(database, database.list_collection_names({}), collections);
+                } else {
+                    config->collectionsFromNameList(database, config->collectionNames, collections);
                 }
-                collections = tempCollections;
             }
 
+            const SteadyClock::time_point started = SteadyClock::now();
+
             // Do each kind of scan.
             if (config->scanType == Count) {
                 countScan(config, collections);
             } else if (config->scanType == Snapshot) {
                 mongocxx::client_session session = _client->start_session({});
                 session.start_transaction(*config->transactionOptions);
                 collectionScan(config, collections);
+                // If a scan duration was specified, we must make the scan
```

Comment:
Block comment?

Reply:
We'll keep multiline comments using '//' as it's more c++ and mongodb style.